File: day-1/typescript/2.ts
let leftList = []
let rightList = []
let similaritySum = 0;

const text = await Deno.readTextFile("input-1.txt").then((text: string) => text.toString())

text.split("\n").forEach((line: string) => {
    const twoNumbers = line.split("   ");

    leftList.push(parseInt(twoNumbers[0]));
    rightList.push(parseInt(twoNumbers[1]));
});

for (let i = 0; i < leftList.length; i++) {
    let timesFound = 0;

    for (let j = 0; j < rightList.length; j++) {
        if (leftList[i] === rightList[j]) {
            timesFound++;
        }
    }

    similaritySum += leftList[i] * timesFound;
}

console.log(similaritySum)

File: day-1/python/2.py
left_list = []
right_list = []
similarity_sum = 0

data = open("../day-1.input", "r")

for line in data:
    left, right = line.split()

    left_list.append(int(left))
    right_list.append(int(right))

data.close()

for i in range(len(left_list)):
    times_found = 0

    for j in range(len(right_list)):
        if left_list[i] == right_list[j]:
            times_found += 1

    similarity_sum += left_list[i] * times_found

print(similarity_sum)